packages/@glimmer/runtime/lib/compiled/opcodes/partial.ts

```diff
@@ -30,11 +28,12 @@ APPEND_OPCODES.add(Op.InvokePartial, (vm, { op1: _meta, op2: _symbols, op3: _eva
     let partialSymbols = symbolTable.symbols;
     let outerScope = vm.scope();
     let partialScope = vm.pushRootScope(partialSymbols.length, false);
+    let evalScope = outerScope.getEvalScope();
     partialScope.bindCallerScope(outerScope.getCallerScope());
-    partialScope.bindEvalScope(outerScope.getEvalScope());
+    partialScope.bindEvalScope(evalScope);
     partialScope.bindSelf(outerScope.getSelf());
 
-    let locals = dict<VersionedPathReference<Opaque>>();
+    let locals = Object.create(outerScope.getPartialMap());
```

Comment:
Actually, I just realized this would turn make `locals` `any` (can you confirm?). We should probably write `as dict<VersionedPathReference<Opaque>>` here (or do an assign like you did, etc, anything works as long as it has the right type).

Reply:
updated to 

`let locals = Object.create(outerScope.getPartialMap()) as Dict<VersionedPathReference<Opaque>>;`